### ISSUE TITLE
Parse error on 'role' word

### ISSUE THREAD

**Issue by TachoMex** [ndmitchell/hlint#278]
I was coding something like 
```haskell
module BugTracer
where
  data UserRole = UserAdmin | UserModerator | RegularUser deriving (Read, Show)

  data User = User {
    name :: String,
    password :: String,
    role :: UserRole
  }
```

and then I ran hlint and got

```
../BugTracer.hs:8:5: Error: Parse error: role
Found:
        name :: String,
        password :: String,
  >     role :: UserRole
      }
  

1 hint 
```
It complains about the word 'role', because if I change it to 'rol' or something else, the hint disappears.

I'm running v1.9.35 over Ubuntu 16.04
```
hlint -V 
HLint v1.9.35, (C) Neil Mitchell 2006-2016
``` 

Thanks!

**Comment by mpickering** [comment]
I think if you upgrade then this will be fixed.

**Comment by TachoMex** [comment]
Sorry, I saw 1.9.35 as current version, but that was for stackage, over hackage it was 1.9.37, and it was solved. Thanks. Sorry for the trouble.
